doc: makeStyles documentation (#17726)
microsoft/fluentui: packages/react-make-styles/README.md

**React bindings for `makeStyles()` for [Fluent UI React](https://developer.microsoft.com/en-us/fluentui)**

These are not production-ready modules and **should never be used in product**. This space is useful for testing new components whose APIs might change before final release.

## `makeStyles()`

> Is used to defined styles, returns a React hook that should be called inside a component:

Reviewer: ```suggestion
Defines a react hook that will return declared styles as classes
```

There are cases where you need to merge classes from multiple `useStyles` calls. To properly merge the classes, you need to use `ax()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.

```tsx
import { ax, makeStyles } from '@fluentui/react-components';

const useStyles = makeStyles({
  blueBold: {
    color: 'blue',
    fontWeight: 'bold',
  },
  red: {
    color: 'red',
  },
});

function Component() {
  const classes = useStyles();

  const firstClassName = ax(classes.blueBold, classes.red); // { color: 'red', fontWeight: 'bold' }
  const secondClassName = ax(classes.red, classes.blueBold); // { color: 'blue', fontWeight: 'bold' }

  return (
    <>
      <div className={firstClassName} />
      <div className={secondClassName} />
    </>
  );
}
```

### 📃 pseudo & class selectors, at-rules, global styles

`makeStyles()` supports pseudo, class selectors and at-rules.

```ts
import { makeStyles } from '@fluentui/react-components';

const useStyles = makeStyles({
  root: {
    ':active': { color: 'pink' },
    ':hover': { color: 'blue' },
    // :link, :focus, etc.

    '.foo': { color: 'black' },

>     '&:nth-child(2n)': { background: '#fafafa' },

Reviewer 2: ```suggestion
    ':nth-child(2n)': { background: '#fafafa' },
```

nit: it's redundant there
